nav2_amcl/src/amcl_node.cpp

```diff
@@ -807,7 +807,16 @@ bool AmclNode::updateFilter(
   RCLCPP_DEBUG(
     get_logger(), "Laser %d angles in base frame: min: %.3f inc: %.3f", laser_index, angle_min,
     angle_increment);
-
+  
+  // Check the validity of range_max, must > 0.0
+  if (laser_scan->range_max <= 0.0){
```

Comment:
I'd say `<` only, because the lines below on L821 check for the laser scan vs parameter max to use. Its only critically misconfigured if its negative

Reply:
So in fact if laser_scan get a wrong value, then there's nothing with the laser_max_range of user input, I think.

Reply:
Fair enough.